tools/opsimUtils.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+
+
+def calc_m5(visitFilter, filtsky, FWHMeff, expTime, airmass, tauCloud=0):
+    """Calculate the m5 value, using opsim methods."""
+    # Set up expected extinction (kAtm) and m5 normalization values (Cm) for each filter.
+    # The Cm values must be changed when telescope and site parameters are updated.
+    #
+    # These values are calculated using $SYSENG_THROUGHPUTS/python/calcM5.py.
+    # This set of values are calculated using v1.2 of the SYSENG_THROUGHPUTS repo.
+    Cm = {'u': 22.74,
+          'g': 24.38,
+          'r': 24.43,
+          'i': 24.30,
+          'z': 24.15,
+          'y': 23.70}
+    dCm_infinity = {'u': 0.75,
+                    'g': 0.19,
+                    'r': 0.10,
+                    'i': 0.07,
+                    'z': 0.05,
+                    'y': 0.04}
+    kAtm = {'u': 0.50,
+            'g': 0.21,
+            'r': 0.13,
+            'i': 0.10,
+            'z': 0.07,
+            'y': 0.18}
+    msky = {'u': 22.95,
+            'g': 22.24,
+            'r': 21.20,
+            'i': 20.47,
+            'z': 19.60,
+            'y': 18.63}
+    # Calculate adjustment if readnoise is significant for exposure time
+    # (see overview paper, equation 7)
+    Tscale = expTime / 30.0 * np.power(10.0, -0.4*(filtsky - msky[visitFilter]))
```

Comment:
I will have to have another look with all the refs out, but I think that it's not necessarily zero, even with 30s exposures. If the sky brightness is very faint, the readnoise can still impact the limiting mag, and this impact is what is tracked by dCm. Zeljko made an error in one of his docs at some point. I'll try to track down our email exchange. (This should match the code in syseng_throughputs, btw).